packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -865,11 +951,16 @@ type CreateTableResponse = {
  * @param event Event for onEvent or isComplete
  * @returns The table input for Custom dynamoDB Table
  */
-export const extractTableInputFromEvent = (
+export const extractTableInputFromEvent = async (
   event: AWSCDKAsyncCustomResource.OnEventRequest | AWSCDKAsyncCustomResource.IsCompleteRequest,
-): CustomDDB.Input => {
+  context: TableManagerContext,
+): Promise<CustomDDB.Input> => {
   // isolate the resource properties from the event and remove the service token
-  const resourceProperties = { ...event.ResourceProperties } as Record<string, any> & { ServiceToken?: string };
+  const tags = await getLambdaTags(context.invokedFunctionArn);
```

Comment:
Can you confirm that the lifecycle of the CDK deployment is such that the Lambda tags are fully deployed by the time this handler is invoked? In other words, we're not going to hit some kind of weird race condition in future where we see tags have not yet been propagated to the function?

I'm assuming this is safe, since Tags are applied during synth, and I would expect them to be applied to the Lambda resource just like other attributes, but since Tags are standalone, we want to make sure we don't hit either lifecycle ordering or eventual consistency problems down the road.

Reply:
I wasn't able to find any documentation on this. I think it would come down to how the custom resource provider is implemented by cloud formation. I will check with someone on cloud formation to see if they know.

Reply:
CFN states the tags will be set before the event on the lambda is processed.